app.go

```
@@ -223,19 +197,16 @@ func (app *App) registerLocationForScope(methods []string, path string, scope Sc
 
 // registerLocationForHost registers a location for a specified hostname.
 func (app *App) registerLocationForHost(methods []string, path, host string, middlewares []middleware.Middleware) {
-	location := vulcan.NewLocation(host, methods, path, app.Config.Name, middlewares)
-
-	if err := app.registry.RegisterLocation(location); err != nil {
-		log.Errorf("Failed to register a location: %v %v", location, err)
-		return
-	}
-
-	if err := app.registry.RegisterFrontend(location); err != nil {
-		log.Errorf("Failed to register a frontend: %v %v", location, err)
-		return
+	registration := &registry.HandlerRegistration{
+		Name:        app.Config.Name,
+		Host:        host,
+		Path:        path,
+		Methods:     methods,
+		Middlewares: middlewares,
 	}
+	app.Config.Registry.RegisterHandler(registration)
```

I need to re-add logging here and stop ignoring this error.